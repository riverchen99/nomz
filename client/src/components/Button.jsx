import React from 'react';
import { StyledButton } from './StyledButton';

function Button(props) {
  const { text, color, handleClick } = props;
  return (
    <StyledButton color={color} onClick={handleClick}>{text}</StyledButton>
  )
}

export default Button;
