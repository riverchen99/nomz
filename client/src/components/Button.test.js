import React from 'react';
import { shallow, mount, configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import Button from './Button';

configure({ adapter: new Adapter() });

const clickFn = jest.fn();
describe('Button component', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<Button />);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with props', () => {
    const component = mount(<Button color="#aaaaaa" text="hi" onClick={clickFn} />);
    expect(component.text()).toEqual('hi');
    expect(component.props().color).toEqual('#aaaaaa');
  });
  it('calls onClick properly', () => {
    const component = mount(<Button color="#aaaaaa" text="hi" onClick={clickFn} />);
    // component.simulate('click'); apparently this doesn't work
    component.props().onClick();
    expect(clickFn).toHaveBeenCalled();
  });
});
