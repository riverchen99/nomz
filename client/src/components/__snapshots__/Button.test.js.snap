// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render correctly with no props 1`] = `ShallowWrapper {}`;
